routing/router.go

```diff
@@ -17,9 +15,42 @@ import (
 	"github.com/roasbeef/btcd/wire"
 	"github.com/roasbeef/btcutil"
 
+	"github.com/go-errors/errors"
 	"github.com/lightningnetwork/lightning-onion"
 )
 
+// Router is responsible for managing the lightning network topology:
+// addition of nodes, edges and applying edges updates.
+type Router interface {
```

Comment:
Hmm, I don't think that `Router` is a good name for this interface. This interface is concerned mostly with storing additional edges+vertexes within the graph and querying for the information in an interactive fashion.

Naming suggestions: `ChannelGraphSource`, `ChannelGraphDatabase`, something along those lines.